services/src/main/scala/cromwell/pipeline/service/UserService.scala

```diff
@@ -31,31 +33,63 @@ trait UserService {
 
 object UserService {
 
+  object Exceptions {
+    sealed abstract class UserServiceException(message: String) extends ServiceException(message)
+
+    final case class AccessDenied(message: String = "Access denied") extends UserServiceException(message)
+    final case class NotFound(message: String = "User with this id doesn't exist") extends UserServiceException(message)
+    final case class WrongEnteredPassword(message: String = "Users password differs from entered")
+        extends UserServiceException(message)
+    final case class WrongRepeatedPassword(message: String = "New password incorrectly duplicated")
+        extends UserServiceException(message)
+    final case class InternalError(message: String = "Internal error") extends UserServiceException(message)
+  }
+
   def apply(userRepository: UserRepository)(implicit executionContext: ExecutionContext): UserService =
     new UserService {
 
       def getUsersByEmail(emailPattern: String): Future[Seq[User]] =
-        userRepository.getUsersByEmail(emailPattern).map(seq => seq.map(User.fromUserWithCredentials))
+        userRepository.getUsersByEmail(emailPattern).map(seq => seq.map(User.fromUserWithCredentials)).recoverWith {
+          case _: Exception =>
+            Future.failed(InternalError("Failed to find user due to unexpected internal error"))
+        }
 
       def getUserWithCredentialsByEmail(email: UserEmail): Future[Option[UserWithCredentials]] =
-        userRepository.getUserByEmail(email)
+        userRepository.getUserByEmail(email).recoverWith {
+          case _: Exception =>
+            Future.failed(InternalError("Failed to find user due to unexpected internal error"))
+        }
 
-      def addUser(user: UserWithCredentials): Future[UserId] = userRepository.addUser(user)
+      def addUser(user: UserWithCredentials): Future[UserId] = userRepository.addUser(user).recoverWith {
+        case _: Exception =>
+          Future.failed(InternalError("Failed to add user due to unexpected internal error"))
+      }
 
-      def deactivateUserById(userId: UserId): Future[Option[User]] =
-        for {
+      def deactivateUserById(userId: UserId): Future[User] = {
+        val deactivatedUser = for {
           _ <- userRepository.deactivateUserById(userId)
           user <- userRepository.getUserById(userId)
```

Comment:
re-use `private def getUserById(userId: UserId): Future[UserWithCredentials]` here

Reply:
Yes, thanks for catch this :)

Reply:
Done.